sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala

```diff
@@ -374,6 +374,52 @@ case class ShowDatabasesCommand(databasePattern: Option[String]) extends Runnabl
 }
 
 /**
+ * A command for users to list the properties for a table.
+ * If propertyKey is specified, the value for the propertyKey
+ * is returned. If propertyKey is not specified, all the keys
+ * and their corresponding values are returned.
+ * The syntax of using this command in SQL is:
+ * {{{
+ *   SHOW TBLPROPERTIES table_name[('propertyKey')];
+ * }}}
+ */
+case class ShowTablePropertiesCommand(
+    table: TableIdentifier,
+    propertyKey: Option[String]) extends RunnableCommand {
+
+  override val output: Seq[Attribute] = {
+    val withKeySchema: Seq[Attribute] = {
+      AttributeReference("value", StringType, nullable = false)() :: Nil
+    }
+    val noKeySchema: Seq[Attribute] = {
+      AttributeReference("key", StringType, nullable = false)() ::
+        AttributeReference("value", StringType, nullable = false)() :: Nil
+    }
+    propertyKey.map(p => withKeySchema).getOrElse(noKeySchema)
```

Comment:
I think this is easier to understand by doing a pattern match of the `propertyKey`.